### ISSUE TITLE
Manifest Merging failing when compiling against Android 12

### ISSUE THREAD

**Issue by MattWilliams89** [urbanairship/android-library#181]
# Preliminary Info

### What Airship dependencies are you using?
14.0.0

### What are the versions of any relevant development tools you are using?

compileSdkVersion = 'android-S'
targetSdkVersion = "S"

# Report

### What unexpected behavior are you seeing?
Manifest merging fails due to the `AdmPushReceiver` BroadcastReceiver missing the `android:exported` tag, as per:
https://medium.com/androiddevelopers/lets-be-explicit-about-our-intent-filters-c5dbe2dbdce0

Error Message:

`Apps targeting Android 12 and higher are required to specify an explicit value for `android:exported` when the corresponding component has an intent filter defined. See https://developer.android.com/guide/topics/manifest/activity-element#exported for details.`

### What is the expected behavior?
Manifest merging successfully completes

### What are the steps to reproduce the unexpected behavior?
Change the compileSdkVersion and targetSdkVersions of the host application to Android 12 (S) and build.

### Do you have logging for the issue?
Nothing of any additional use to the information above

**Comment by jyaganeh** [comment]
@MattWilliams89 Thanks for the report. We've got a fix for this internally that should make it into the next release.

In the meantime, manifest merging can be used to work around the error. Here's what you'll need to add to your manifest for `AdmPushReceiver`:

```xml
<receiver android:name="com.urbanairship.push.adm.AdmPushReceiver" 
    android:exported="true" />
``` 

Android Studio will show an "Exported receiver does not require permission" warning, which you can safely ignore if you're just testing Android 12 compatibility. The warning can be suppressed with:
```
tools:ignore="ExportedReceiver"
``` 

or you could add the permission to match the `urbanairship-adm` manifest: 
```
android:permission="com.amazon.device.messaging.permission.SEND"
```

**Comment by jyaganeh** [comment]
@MattWilliams89 We shipped release `14.4.4` today, which includes updates for building with the Android 12 Developer Preview.

The current stable version of WorkManager that we depend on hasn't been updated to make `PendingIntent` mutability explicit yet, so for now you'll need to add the following to your `build.gradle` to force Gradle to use an alpha release of WorkManager for testing purposes.

```groovy
dependencies {
    def airshipVersion = "14.4.4"
    implementation "com.urbanairship.android:urbanairship-fcm:$airshipVersion"
    implementation "com.urbanairship.android:urbanairship-adm:$airshipVersion"
    implementation "com.urbanairship.android:urbanairship-automation:$airshipVersion"
    implementation "com.urbanairship.android:urbanairship-message-center:$airshipVersion"
    // ...

    // Upgrade WorkManager to 2.7-alpha03 for Android 12 testing
    // Note: WorkManager 2.7-alpha03 is only compatible with the Android 12 Developer Preview 1 SDK
    constraints {
        implementation("androidx.work:work-runtime:2.7.0-alpha03")
    }
}
```

We'll be keeping an eye out for updates and will upgrade our WorkManager version when a new stable release is available.
